moesifwsgi/middleware.py

```diff
@@ -103,21 +103,39 @@ def __call__(self, environ, start_response):
                         environ['REQUEST_METHOD'],
                         self.logger_helper.request_url(environ),
                         self.client_ip.get_client_address(environ),
-                        self.logger_helper.get_user_id(environ, self.settings, self.app, self.DEBUG),
-                        self.logger_helper.get_company_id(environ, self.settings, self.app, self.DEBUG),
-                        self.logger_helper.get_metadata(environ, self.settings, self.app, self.DEBUG),
-                        self.logger_helper.get_session_token(environ, self.settings, self.app, self.DEBUG),
                         [(k, v) for k,v in self.logger_helper.parse_request_headers(environ)],
                         *self.logger_helper.request_body(environ)
                     )
 
         def _start_response(status, response_headers, *args):
             # Capture status and response_headers for later processing
             data_holder.capture_response_status(status, response_headers)
+
+            if response_headers:
+                response_headers_dict = {}
+                try:
+                    for pair in response_headers:
+                        response_headers_dict[pair[0].lower()] = pair[1]
+                except Exception as e:
+                    print('Error while parsing response headers', e)
+
+                environ['response-headers'] = response_headers_dict
+
             return start_response(status, response_headers, *args)
 
         response_chunks = data_holder.finish_response(self.app(environ, _start_response))
 
+        try:
+            decoded_response_body = ''.join((x.decode('utf-8') for x in response_chunks))
+            environ['response-body'] = decoded_response_body
```

Comment:
may be we set the key to be `moesif_xx` so that it doesn't conflict with anything downstream?

Reply:
Changed